app/scripts/modules/core/task/task.write.service.js

```diff
@@ -19,11 +20,38 @@ module.exports = angular
     }
 
     function cancelTask(application, taskId) {
-      return getEndpoint(application).one(taskId).one('cancel').put();
+      return getEndpoint(application).one(taskId).one('cancel').put().then(() =>
+        waitUntilTaskMatches(application, taskId, (task) => task.status === 'CANCELED')
+      );
+    }
+
+    function waitUntilTaskMatches(application, taskId, closure) {
+      return taskReader.getOneTaskForApplication(application, taskId).then((task) => {
+        if (closure(task)) {
+          return task;
+        }
+        return $timeout(() => waitUntilTaskMatches(application, taskId, closure), 1000);
+      });
+    }
+
+    function waitUntilTaskIsDeleted(taskId) {
+      var deferred = $q.defer();
+      Restangular.one('tasks', taskId).get().then(
+        () => {
+          $timeout(() => {
+            waitUntilTaskIsDeleted(taskId).then(() => deferred.resolve());
+          }, 1000);
+        },
+        (resp) => {
+          if (resp.status === 404) {
+            deferred.resolve();
+          }
+        });
+      return deferred.promise;
```

Comment:
Is it necessary to manually create the promise here? The $timeout returns a promise that we can just return in the success function.  

This passes the tests... might want to add a catch function on the calling function (deleteTask). 
Not 100% on how to handle the error handling on recursive promises. I'd need to look into that more.

    function waitUntilTaskIsDeleted(taskId) {
      return Restangular.one('tasks', taskId).get().then(
        () => {
          return $timeout(
            () => {
              waitUntilTaskIsDeleted(taskId);
            }, 1000);
        });
    }